cmake/checks/check_02_compiler_features.cmake

```diff
@@ -405,7 +409,11 @@ ENDIF()
 #
 # - Matthias Maier, 2015
 #
-ADD_FLAGS(CMAKE_REQUIRED_FLAGS "-Werror")
+IF(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  ADD_FLAGS(CMAKE_REQUIRED_FLAGS "/WX")
+ELSE()
```

Comment:
This kind of block is repeated many times now. Would it be useful to abstract all of this into a `ADD_Werror_FLAG` or something similar?

Reply:
I am not sure if that is much better, but I did this in the last commit.